selene-lib/src/rules/limit_function_complexity.rs

```diff
@@ -244,10 +244,36 @@ fn count_block_complexity(block: &ast::Block, starting_complexity: u16) -> u16 {
 }
 
 impl Visitor for LimitFunctionComplexityVisitor {
-    fn visit_function_body(&mut self, function_body: &ast::FunctionBody) {
-        let complexity = count_block_complexity(function_body.block(), 1);
+    fn visit_local_function(&mut self, local_function: &ast::LocalFunction) {
+        let complexity = count_block_complexity(local_function.body().block(), 1);
         if complexity > self.config.maximum_complexity {
-            self.positions.push((range(function_body.parameters_parentheses()), complexity));
+            self.positions.push((
+                (range(local_function.function_token()).0, range(local_function.body().parameters_parentheses()).1),
+                complexity
+            ));
+        }
+    }
+
+    fn visit_function_declaration(&mut self, function_declaration: &ast::FunctionDeclaration) {
+        let complexity = count_block_complexity(function_declaration.body().block(), 1);
+        if complexity > self.config.maximum_complexity {
+            self.positions.push((
+                (range(function_declaration.function_token()).0, range(function_declaration.body().parameters_parentheses()).1),
+                complexity
+            ));
+        }
+    }
+
+    fn visit_value(&mut self, value: &ast::Value) {
+        if let ast::Value::Function(function_value) = value {
+            let function_body = &(function_value).1;

```

Comment:
Try any of these to see if they work (in descending order of preference):

```suggestion
        if let ast::Value::Function((_, function_body)) = value {
```

```suggestion
        if let ast::Value::Function(function_value) = value {
            let function_body = function_value.1;
```

```suggestion
        if let ast::Value::Function(function_value) = value {
            let function_body = (&function_value).1;
```